assets/js/src/nps_poll.jsx

```diff
@@ -70,29 +55,35 @@ class NpsPoll extends React.Component {
       events: {
         submit: (response) => {
           if (response.rating >= 9 && response.completed) {
-            this.showReviewRequestModal();
+            showReviewRequestModal();
           }
         },
       },
     });
   }
 
-  render() {
-    if (!window.mailpoet_display_nps_poll) {
-      return null;
-    }
-    if (window.satismeter) {
-      setImmediate(this.displayPoll);
-    } else {
-      const s = document.createElement('script');
-      s.type = 'text/javascript';
-      s.src = 'https://app.satismeter.com/satismeter.js';
-      s.onload = () => this.displayPoll();
-      document.getElementsByTagName('body')[0].appendChild(s);
+  function displayPoll() {
+    if (!pollShown) {
+      setPollShown(true);
+      getTrackingData().then(data => callSatismeter(data.data));
     }
+  }
+
+  if (!window.mailpoet_display_nps_poll) {
     return null;
   }
-}
+
+  if (window.satismeter) {
+    setImmediate(displayPoll);
```

Comment:
I don't think it's a good idea to use `setImmediate` here. First, it's not a standard JS function according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate). Second, it breaks the flow of React which wants the control when things are run and when rendering happens. I think we should use `useLayoutEffect` for this behavior.